src/main/java/de/tum/in/www1/artemis/domain/ProgrammingExercise.java

```diff
@@ -22,6 +24,7 @@
  */
 @Entity
 @DiscriminatorValue(value = "P")
+@SecondaryTable(name = "programming_exercise_details")
```

Comment:
Is the advantage of moving some properties to another table really that big? I get that there are exercise specific values, but the class `PrgrammingExercise` is a specialization in itself (although of course, we still have all columns in one table, which was your reason for adding this annotation in the first place). By introducing this table we add more load on the server since every request fetching a programming exercise will have to join the main table with the secondary. . What is your opinion on this?

Reply:
@ungaralex no join is necessary. Just two select queries instead of one which is handled automatically and which should be minimal additional effort. The new table uses the same primary key. When hibernate can use the cache (because the exercise was loaded before), there is no additional effort at all. The big advantage is that the `exercise` table does not get bigger and bigger due to options specific to only one exercise type (we already have many null values and will migrate more of them into secondary tables specific to only one exercise type) 
And in the future we might be able to leave out querying certain properties in certain places. For example the new option in this PR is not necessary in the exercise dashboard for students and then one select might be fine as well.

Reply:
Ah, great. Thanks for the clarification!